components/column-linking-table.vue

```diff
@@ -18,23 +19,138 @@
 
 <script>
 
+    // Allows for reference to store data by creating simple, implicit getters
+    // Fields listed in mapState below can be found in the store (index.js)
+    import { mapState } from "vuex";
+
+    // Allows for reference to store actions (index.js)
+    import { mapActions } from "vuex";
+
     export default {
 
         props: {
 
-            categoryClasses: { type: Object, required: true },
-            columnToCategoryMap: { type: Object, required: true },
-            fields: { type: Array, required: true },
-            selectedCategory: { type: String, required: true },
-            tableData: { type: Array, required: true }
+            selectedCategory: { type: String, required: true }
+        },
+
+        data() {
+
+            return {
+
+                columnToCategoryTable: [],
+
+                uiText: {
+
+                    tableFields: [
+
+                        { key: "column" },
+                        { key: "description" }
+                    ]
+                }
+            };
+        },
+
+        created() {
+
+            // Create the data structure for the category to column linking table
+            this.setupColumnToCategoryTable();
+        },
+
+        computed: {
+
+            ...mapState([
+
+                "categoryClasses",
+                "columnToCategoryMap",
+                "dataDictionary",
+                "dataTable"
+            ])
         },
 
         methods: {
 
+            ...mapActions([
+
+                "linkColumnWithCategory",
+                "unlinkColumnFromCategory"
+            ]),
+
             applyCategory(p_row, p_index, p_event) {
 
-                // Tell the parent page that a column has been linked with a category
-                this.$emit("column-name-selected", { column: p_row.column });
+                const dataForStore = { column: p_row.column };
+
+                // 1. Link or unlink the currently-selected category and the clicked column
+                if ( this.selectedCategory !== this.columnToCategoryMap[p_row.column] ) {
+
+                    this.linkColumnWithCategory({...dataForStore, category: this.selectedCategory});
+                } else {
+
+                    this.unlinkColumnFromCategory(dataForStore);
+                }
+
+                // 2. Tell the categorization page a column has been clicked
+                this.$emit("column-name-clicked", dataForStore);
+            },
+
+            setupColumnToCategoryTable() {
```

Comment:
I get what this method does. But given how many assumptions we're making here about how the store will be organized, I think it might be cleaner to just ask the store to provide the data elements we want here directly - like we're doing in other components. E.g. one getter to get the column names of the data table, and a second getter to get the description for each column if there is one. 

Then we don't have to do so much work in the component and also have an easier time if we decide to change the specifics in the store.

Reply:
As per our discussion of refactoring/writing tests toward an ideal store, this function is removed. Its logic will be split apart/reworked and added to the store. (This work will be added as a subtask of the new store refactor issue I will be creating.)

Instead, now the table rows are computed property that returns an array of objects with keys for column name, column description, and category. The first two pull from a to-be-implemented `columns` store getter that returns an array of objects containing a column name and description per array entry.

The unused column of `category` has also been added to the table (and is now utilized) as another visual indicator of the link between column and category. See `data().uiText.tableFields`.